src/github.com/matrix-org/dendrite/clientapi/auth/storage/accounts/storage.go

```diff
@@ -314,3 +319,21 @@ func (d *Database) GetThreePIDsForLocalpart(
 ) (threepids []authtypes.ThreePID, err error) {
 	return d.threepids.selectThreePIDsForLocalpart(ctx, localpart)
 }
+
+// GetFilter looks up the filter associated with a given local user and filter ID.
+// Returns an error if no such filter exists or if there was an error taling to the database.
+func (d *Database) GetFilter(
+	ctx context.Context, localpart string, filterID string,
+) (string, error) {
+	return d.filter.selectFilter(ctx, localpart, filterID)
+}
+
+func (d *Database) PutFilter(
+	ctx context.Context, localpart, filter string,
+) (string, error) {
+	err := d.filter.insertFilter(ctx, filter, localpart)
+	if err != nil {
+		return "", err
+	}
+	return d.filter.findMaxID(ctx, localpart)
```

Comment:
This is somewhat ugly, but I haven't found another way to get the integer created by postgres.

Reply:
You can use ```RETURNING``` to get the ID chosen by the ```INSERT``` https://www.postgresql.org/docs/9.5/static/dml-returning.html
There's an example of use doing this in dendrite at https://github.com/matrix-org/dendrite/blob/master/src/github.com/matrix-org/dendrite/syncapi/storage/output_room_events_table.go#L56

Reply:
I'll definitely do this, if we stick to using "SERIAL" here. If we use the hash as the ID, this is would not be needed anymore.